corehq/apps/reports/filters/users.py

```diff
@@ -343,10 +349,14 @@ def pull_users_and_groups(cls, domain, mobile_user_and_group_slugs,
 
         user_dict = {}
         for group in groups:
-            user_dict["%s|%s" % (group.name, group._id)] = util.get_all_users_by_domain(
+            users_in_group = util.get_all_users_by_domain(
                 group=group,
                 simplified=True
             )
+            if limit_user_ids:
+                users_in_group = filter(lambda user: user['user_id'] in limit_user_ids, users_in_group)
+            user_dict["%s|%s" % (group.name, group._id)] = users_in_group
```

Comment:
yeah wanted to stay away from doing any larger refactors in the PR, but agree this function could use some love